features/gezag-minderjarige/dev/1.1-ingezetene-BRP-dev.feature

```diff
@@ -0,0 +1,147 @@
+#language: nl
+
+Functionaliteit: 1.1 - Staat persoon (minderjarige) als ingezetene in de BRP?
+
+  De persoon (minderjarige) moet voorkomen in de BRP aangezien dit de bron is voor
+  het achterhalen van het gezag. Een niet ingezeten kind staat wel de RNI (ook
+  onderdeel van de BRP), maar hiervan zijn de gegevens (mogelijk) niet actueel. Een
+  gezagsmutatie wordt bijvoorbeeld niet bijgewerkt in de RNI.
+
+  Een persoon is "ingezetene" en "ingeschreven in de BRP" wanneer deze staat ingeschreven
+  in een Nederlandse gemeente. In dat geval gaat de gezagbepaling 
+  verder met vraag 1.2.
+
+  Een persoon is "niet ingezetene" wanneer deze staat ingeschreven in het RNI (Register Niet 
+  Ingezetenen). In dat gezal is het gezag niet te bepalen.
+
+
+    Achtergrond:
+      Gegeven de persoon 'Gerda' met burgerservicenummer '000000012'
+      * is meerderjarig
+      En de persoon 'Aart' met burgerservicenummer '000000024'
+      * is meerderjarig
+      En 'Gerda' en 'Aart' zijn met elkaar gehuwd
+      En de persoon 'Zoe' met burgerservicenummer '000000036'
+      * is minderjarig
+      * is in Nederland geboren
+      * heeft 'Gerda' als ouder 1
+      * heeft 'Aart' als ouder 2
+
+
+  Regel: Als de gemeente van inschrijving is RNI kan het gezag van de persoon niet worden bepaald
+    Een persoon staat in de RNI ingeschreven wanneer gemeente van inschrijving (09.10) gelijk is aan 1999.
+
+    Dit wordt bepaald in tabel lo3_pl_verblijfplaats, kolom inschrijving_gemeente_code, waarbij volg_nr = 0.
+    Wanneer inschrijving_gemeente_code = 1999 is gezag niet te bepalen.
+
+    Scenario: minderjarige staat ingeschreven in RNI
```

Comment:
Ik weet niet of het handig is om een scenario op twee verschillende manieren (één voor domein expert en één voor API developer) te specificeren. In principe krijgt je één scenario die alle implementatie detail verbergt en één scenario waarbij alle niet-relevante implementatie detail wordt weggelaten. Dit betekent dan dat de automation moet bepalen welke niet-relevante implementatie detail is weggelaten en deze moet toevoegen zodat de gewenste insert niet zal falen

Reply:
Zijn er bezwaren om uit te leggen wat 'ingeschreven in RNI' betekent met de scenarios op regel 41 en regel 60 in de volgende feature file: https://github.com/BRP-API/brp-shared/blob/893ddc15cd5a085938d0f55bcdc732b9a681c9be/features/docs/integration/gegeven-stap-definities-persoon.feature.

Deze scenarios borduurt voort op de pull request die ik afgelopen maandag hebt gemaakt om de gegeven stappen te documenteren en te gebruiken als integratie tests (wordt er uit de gegeven stappen insert statements gegenereerd die ook daadwerkelijk kunnen worden uitgevoerd)

Reply:
> Zijn er bezwaren om uit te leggen wat 'ingeschreven in RNI' betekent met de scenarios op regel 41 en regel 60 in de volgende feature file: https://github.com/BRP-API/brp-shared/blob/893ddc15cd5a085938d0f55bcdc732b9a681c9be/features/docs/integration/gegeven-stap-definities-persoon.feature.
> 
> Deze scenarios borduurt voort op de pull request die ik afgelopen maandag heb gemaakt om de gegeven stappen te documenteren en te gebruiken als integratie tests (wordt er uit de gegeven stappen insert statements gegenereerd die ook daadwerkelijk kunnen worden uitgevoerd)

Ik weet niet of ik je vraag goed begrijp. Ik zie de uitleg van gegeven stap "ingeschreven in RNI" in de feature waar je naar verwijst als iets anders dan de uitleg van "ingeschreven in RNI" in deze gezag feature.

Ik denk dat elke feature voor gezag in essentie alleen gaat over definities en hoe die technisch bepaald worden:
- vraag 1.1: "ingeschreven in RNI" en "ingeschreven in BRP"
- vraag 1.2: "is minderjarig" en "is meerderjarig" en "is overleden"
- vraag 1.3: "is geëmigreerd geweest"
- vraag 1.3a: "is in Nederland geboren" en "is in het buitenland geboren"
- vraag 1.3b: "is geadopteerd"
- enz.

Reply:
> 
> Ik weet niet of ik je vraag goed begrijp. Ik zie de uitleg van gegeven stap "ingeschreven in RNI" in de feature waar je naar verwijst als iets anders dan de uitleg van "ingeschreven in RNI" in deze gezag feature.
> 
Met de p.r. '[documenteren gegeven stap definities als integratie tests](https://github.com/BRP-API/brp-shared/pull/12)' heb ik juist geprobeerd te verduidelijken wat de expressieve stappen technisch betekenen. Maar dat is dus niet aangekomen.

Ik weet niet hoe ik mijn bezwaar moet uitleggen, maar misschien helpt het als ik het designen van de gegeven stappen vergelijk met het designen van een API, want zoals ik al eerder heb aangegeven kunnen de gegeven stappen worden gezien als de operaties van de BRP mutatie API.

Bij de expressieve variant heb je voor de 'ingeschreven in RNI' en de 'ingeschreven in BRP' mutaties twee endpoints nodig:
'IngeschrevenInDeBrp' en 'IngeschrevenInRni' met als parameters de bsn van de persoon die moet worden ingeschreven en een boolean om aan te geven of de inschrijving een correctie is van de vorige.

Bij de dev variant die jij nu voorstel is er maar één operatie 'muteerVerblijfplaats'. Naast de bsn van de persoon geeft je een array (rijen) met codes (magic numbers) hoe de mutaties moet worden uitgevoerd. Voordeel van deze variant is dat je in één aanroep alle verschillende varianten van mutaties (eerst ingeschreven in de RNI en daarna in de BRP en omgekeerd in combinatie met correcties), maar je moet wel weten wat de codes betekenen en in welke combinatie je de codes wel/niet kan gebruiken.

Ik hoop dat ik het nu aan je verbeelding kan overlaten om te bedenken bij welke gegeven stappen de developers straks er het meeste werk aan hebben

Reply:
Persoonlijk denk ik dat bovenstaande de features te ingewikkeld maakt.

Er is hier eigenlijk sprake van een binaire test: wel ingeschreven in de BRP, niet ingeschreven in de BRP.

Mogelijk moet de stap `is ingeschreven in RNI` vervallen en moeten er twee gegeven stappen zijn:
`is ingeschreven in de BRP` -> is er al
`is niet ingeschreven in de BRP` -> die doet op de achtergrond de gemeente van inschrijving voor de verblijfplaats aanpassen naar 1999


Dat niet ingeschreven in de BRP betekend dat gemeente van inschrijving 1999 is kan naar de developers informatie wanneer we daar een vorm voor bedacht hebben volgens mij, dat is een implementatie detail toch.

Zie ook mijn comment onderaan.